algebras/algebra/src/test/scala/endpoints4s/algebra/server/AssetsTestSuite.scala

```diff
@@ -0,0 +1,89 @@
+package endpoints4s.algebra.server
+
+import akka.http.scaladsl.model.HttpMethods.{GET}
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.headers.{`Last-Modified`, `If-Modified-Since`, `Content-Type`}
+import akka.http.scaladsl.model.DateTime
+import akka.http.scaladsl.model.ContentType
+import akka.http.scaladsl.model.MediaTypes
+
+trait AssetsTestSuite[T <: endpoints4s.algebra.AssetsTestApi] extends ServerAssetTest[T] {
```

Comment:
I followed your suggestion (https://github.com/endpoints4s/endpoints4s/pull/607/commits/19d5432dcdc6129c719ecd801ee452850dfb1127), and this indeed seems to be a good call. The result is that much of the behavior of the `Asset` algebra can now indeed by tested in the algebra test. Also note that with these tests, the play interpreter would need a little bit rework in order to pass these tests; that's a good thing probably as it means all interpreters will converge to a singular behavior.

One remark: I created a separate `ServerAssets` algebra, since otherwise I would have needed to implement the method to construct an `AssetResponse` (`foundAssetResponse`) as well for the xhr interpreter, something that did not make sense to me.

Reply:
> I followed your suggestion ([19d5432](https://github.com/endpoints4s/endpoints4s/commit/19d5432dcdc6129c719ecd801ee452850dfb1127)), and this indeed seems to be a good call. The result is that much of the behavior of the `Asset` algebra can now indeed by tested in the algebra test

Sorry, I think I wasn’t clear but I was thinking of introducing the `AssetResponse` constructors in the **test** (in `endpoints4s.algebra.server`) module, not the main module, so that we don’t have to change the algebra itself.

Reply:
That makes a lot of sense actually :sweat_smile: , updated in https://github.com/endpoints4s/endpoints4s/pull/607/commits/16353c0b22727db4d911c06ab05997cc52bdc201